src/server/acl/acl_commands_def.h

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include "absl/container/flat_hash_map.h"
+
+namespace dfly {
+namespace AclCategory {
```

Comment:
I am not a big fan of introducing lots of namespaces because it  eats up horizontal space and hurts eyes.

1. if you want to add a namespace, then it's better to make it short and consistent with the folder name (i.e. acl).
2. everything in the folder should be under the same namespace. 
3. classes inside the namespace should not have Acl prefix - i.e. either with namespace and without Acl or without the namespace with prefix.

Reply:
+1 . note that I introduced the acl namespace here just so the enum constants in `enum AclCat` don't leak (since it's an unscoped enum). A class enum would not do either, because each time you assign one of its constants to a variable you need to `static_cast`. The last one was `constexpr` variables (which was originally what I did) but later Vlad pointed out that grouping with the enum is nice and I ended up opting for `namespace + enum`.

I will put everything in `acl` subfolder under the same namespace `acl`. 

Just to clarify, the acl should be a top level namespace, that is `::acl::some_def` and not `dfly::acl::some_def` is that correct?

Reply:
everything is `dfly` under `src/server`, therefore I think it makes sense to introduce `dfly::acl`. 
`facade` is the only other top-level namespace we introduced in dragonfly repo, imho.